File: front/src/components/Card.js
import React from "react";

const map = new Map([
    ["TWO", "2"],
    ["THREE", "3"],
    ["FOUR", "4"],
    ["FIVE", "5"],
    ["SIX", "6"],
    ["SEVEN", "7"],
    ["EIGHT", "8"],
    ["NINE", "9"],
    ["TEN", "10"],
    ["JACK", "J"],
    ["QUEEN", "Q"],
    ["KING", "K"],
    ["ACE", "A"],
    ["CLUBS", "♣"],
    ["DIAMONDS", "♦"],
    ["HEARTS", "♥"],
    ["SPADES", "♠"],
    ]);

const colorMap = new Map([
    ["CLUBS", "black"],
    ["DIAMONDS", "red"],
    ["HEARTS", "red"],
    ["SPADES", "black"],
    ]);

function Card({ card }) {
  return (
  <div style={{ color: colorMap.get(card.suit) ?? 'black', border: `1px solid ${colorMap.get(card.suit) ?? 'black'}`, padding: '5px', margin: '5px' }}>
    {map.get(card.value)?? ''}{map.get(card.suit) ?? ''}
  </div>);
}

export default Card;

File: back/src/main/java/com/job/cards/model/Card.java
package com.job.cards.model;

public class Card {

    private Suit suit;
    private Value value;

    public Card(Suit suit, Value value) {
        this.suit = suit;
        this.value = value;
    }

    public Suit getSuit() {
        return suit;
    }

    public Value getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value + " of " + suit;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof Card) {
            Card card = (Card) obj;
            return suit == card.suit && value == card.value;
        }
        return false;
    }

    public enum Suit {
        CLUBS, DIAMONDS, HEARTS, SPADES
    }

    public enum Value {
        ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING
    }
}
